### DISCUSSION TITLE
Are there any ways to programmatically retrieve the duration of a job?

### DISCUSSION THREAD

**Discussion by iambenkay** [gocd/gocd#10784] [Q&A]
I have a use case where I need to send the duration of a job to an external API for record keeping purposes. Is there any way I can get the time that has been spent running the job from within it? maybe as an environment variable. Is this data being tracked?

Getting this duration data after the job has been run is also an alternative I am willing to live with

**Comment by arvindsv** [comment]
You can use the Job Instance API: https://api.gocd.org/current/#get-job-instance

The API's time format is inconsistent with the rest of the API and a new version of that API method is being discussed: https://github.com/gocd/gocd/issues/10713 -- but what is documented there should be enough to get the duration with some calculation.

By the way, it's likely that more people monitor the [GoCD mailing list](https://groups.google.com/forum/#!forum/go-cd) than GitHub discussions. It's much easier for these questions to get lost in GitHub Discussions.
